exo-run/src/docker-runner.ls

```diff
@@ -63,11 +65,20 @@ class DockerRunner extends EventEmitter
         @emit 'online'
 
 
-  _check-dependency-containers: ~>
-    for dependency of @docker-config.dependencies
-      app-dependency = "#{@docker-config.app-name}-#{dependency}"
-      DockerHelper.ensure-container-is-running app-dependency, dependency
-      @docker-config.env[dependency.to-upper-case!] = DockerHelper.get-docker-ip app-dependency
+  _check-dependency-containers: (done) ~>
+    dependencies = []
+    for image, vars of @docker-config.dependencies
+      container-name = "#{@docker-config.app-name}-#{image}"
+      if vars.docker_flags?
+        volume = Handlebars.compile(that.volume)({"EXO_DATA_PATH": "/Users/originate/Desktop/data"})
+        online-text = that.online_text
+      dependencies.push {container-name, image, volume, online-text}
+
+    async.map-series dependencies, DockerHelper.ensure-container-is-running, (err) ~>
+      | err  =>  @emit 'error', err
```

Comment:
Okay, good for now. We have to re-think the overall architecture of this, but that can be done later.